src/context.rs

```diff
@@ -26,6 +26,22 @@ impl Context {
         }
     }
 
+    #[cfg(feature = "experimental-api")]
+    pub fn get_thread_safe_context() -> Self {
+        let ctx = unsafe { raw::RedisModule_GetThreadSafeContext.unwrap()(ptr::null_mut()) };
+        Context::new(ctx)
+    }
+
+    #[cfg(feature = "experimental-api")]
+    pub fn lock(&self) {
```

Comment:
It would be nice to model this like `std::sync::Mutex`:
- `get_thread_safe_context` would return a guard object, instead of the `Context` itself.
- Calling `lock` in the guard object would return the `Context` stored inside the guard.
- `impl Drop` on the guard would unlock the `Context`.